### PR TITLE
Vault refactor

### PR DESCRIPTION
**Pull request by derek-pryor**
Refactored the Vault deployment for the BOSS:
* Replaced the Consul storage backend with the DynamoDB storage backend, allowing us to remove the Consul cluster
* Moved to using the AWS KMS seal configuration to store the master encryption key in KMS, allowing for the Autoscale Group to terminate and restart unhealthy instances
* Repurposed the Vault health check lambda to set the ASG instance health instead of sending a SNS message for manual intervention
* Updated the CloudFormation core configuration update method to migrate data between the old and new Vault deployments
* Updated to use the latested Python `hvac` library (its API changed) so that Vault using the seal configuration could be initialized

Notes:
* Updating core will export the data, destroy the existing Vault cluster, create a new one, initialize it, and import the saved data
* Exported Vault data does **not** include the AWS policies/configuration that is created for each ingest job. If there are ingest jobs defined when running update they will be left in a broken state
* Before running update you will need to install the latested (`0.8.1`) version of `hvac` to be able to initialize Vault.

### CODE REVIEW

**Comment by derek-pryor** [comment]
Once the integration branch is ready I will merge it into this feature branch to resolve the merge conflicts

**Comment by derek-pryor** [comment]
Up-to-date with the integration branch.

I updated the Vault related policies and roles to be more specific to what is needed (based on the Hashicorp documentation) and removed several Vault policies no longer used.

File: lib/vault.py [hunk 1]
```diff
@@ -102,7 +102,7 @@ def shell(self):
         client = self.connect(VAULT_TOKEN)
         code.interact(local=locals())
 
-    def initialize(self, secrets = 5, threashold = 3):
+    def initialize(self, secrets = 1, threashold = 1):
         """Initialize a Vault. Connect using get_client() and if the Vault is not
         initialized then initialize it with 5 secrets and a threashold of 3. The
```
Comment (movestill) [inline lib/vault.py]: This line is out of date now.

File: lib/vault.py [hunk 2]
```diff
@@ -115,34 +115,65 @@ def initialize(self, secrets = 5, threashold = 3):
...
+        #          > local node not active but active cluster node not found <
+        #          If given a little bit of time Vault will respond successfully
+        #          to requests
+
+        def poll():
+            """Check to see if Vault responds to a request without an error"""
+            try:
+                self.connect(VAULT_TOKEN).sys.list_enabled_audit_devices()
+                return True
+            except:
```
Comment (movestill) [inline lib/vault.py]: Do we want to catch specific exceptions so we can fail faster if something unexpected occurs?

File: lib/vault.py (hunk 1 above)
Comment (derek-pryor) [reply lib/vault.py]: Good catch

File: lib/vault.py (hunk 2 above)
Comment (derek-pryor) [reply lib/vault.py]: Updated with both catching the exception and checking the exception message

File: lib/migrations.py [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+        self.filepath = filepath
+
+        module = filepath.replace(const.repo_path(), '')[1:]
+        module = os.path.splitext(module)[0]
+        module = module.replace('/', '.')
+        self.module_path = module
+
+        filename = module.rsplit('.', 1)[1]
+        vers, self.name = filename.split('_', 1)
+        self.start, self.stop = int(vers[0:4]), int(vers[4:8])
```
Comment (movestill) [inline lib/migrations.py]: Is the naming convention for migration files documented anywhere besides what's done in the code here?

File: bin/cloudformation.py [hunk 4]
```diff
@@ -200,7 +229,7 @@ def create_help(header, options):
     config_names = [x.split('/')[1].split('.')[0] for x in glob.glob("configs/*.py") if "__init__" not in x]
     config_help = create_help("config_name supports the following:", config_names)
 
-    actions = ["create", "update", "delete", "post-init", "pre-init", "generate"]
+    actions = ["create", "update", "delete", "post-init", "pre-init", "update-migrate", "generate"]
```
Comment (movestill) [inline bin/cloudformation.py]: Do we want to change the name of the action? Maybe it's easier for the user to just leave it as "update"?

Comment (derek-pryor) [reply bin/cloudformation.py]: The problem is how to determine if they mean run CF update or just the post-update migrations? I could call it `post-update` but I was trying to not take that name if we wanted to support pre-update and post-update like we do pre-init and post-init...

File: lib/migrations.py (hunk 3 above)
Comment (derek-pryor) [reply lib/migrations.py]: Thanks for catching that. I meant to document that

File: bin/cloudformation.py (hunk 4 above)
Comment (movestill) [reply bin/cloudformation.py]: Oh I totally misread the change. I thought I saw `update` removed and replaced with `update-migrate`. Is `post-update-migrate` too long? It does give the user a sense of exactly what's happening, though.

Comment (derek-pryor) [reply bin/cloudformation.py]: Not sure. @sandyhider / @lrodri29 any input into the command name?

File: lib/migrations.py (hunk 3 above)
Comment (derek-pryor) [reply lib/migrations.py]: Added a README in the migrations directory

**Pull request merged**